src/interfaces/vault.spec.ts

```diff
@@ -0,0 +1,868 @@
+    provider: {
+      write: {
+        getSigner: jest.fn().mockImplementation(() => ({
+          sendTransaction: sendTransactionMock
+        }))
+      }
+    }
+  }))
+}));
+
+const PickleJarsMock = jest.requireMock("../services/partners/pickle");
+jest.mock("../services/partners/pickle", () => ({
+  PickleJars: []
+}));
+
+jest.mock("@ethersproject/contracts", () => ({
+  Contract: jest.fn().mockImplementation(() => ({
+    deposit: jest.fn(),
+    withdraw: jest.fn()
+  }))
+}));
+
+describe("VaultInterface", () => {
+  let vaultInterface: VaultInterface<1>;
+
+  let mockedYearn: Yearn<ChainId>;
+
+  beforeEach(() => {
+    mockedYearn = new (Yearn as jest.Mock<Yearn<ChainId>>)();
+    vaultInterface = new VaultInterface(mockedYearn, 1, new Context({}));
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("get", () => {
+    describe("when the fetcher tokens are cached", () => {
+      let cachedToken: Token;
+      let randomCachedToken: Token;
+
+      beforeEach(() => {
+        cachedToken = createMockToken({ address: "0xCachedToken" });
+        randomCachedToken = createMockToken({ address: "0xCachedRandom" });
+        cachedFetcherFetchMock.mockResolvedValue([cachedToken, randomCachedToken]);
+      });
+
+      describe("when the addresses filter is given", () => {
+        it("should get all cached tokens in the addresses array filter", async () => {
+          const actualGet = await vaultInterface.get(["0xCachedToken"]);
+
+          expect(actualGet).toEqual([cachedToken]);
+        });
+      });
+
+      describe("when the addresses filter is not given", () => {
+        it("should get all cached tokens", async () => {
+          const actualGet = await vaultInterface.get();
+
+          expect(actualGet).toEqual([cachedToken, randomCachedToken]);
+        });
+      });
+    });
+
+    describe("when the fetcher tokens are not cached", () => {
+      let getStaticMock: jest.Mock;
+      let vaultDynamic: AssetDynamic<"VAULT_V2">;
+
+      beforeEach(() => {
+        cachedFetcherFetchMock.mockResolvedValue(undefined);
+        metaVaultsMock.mockResolvedValue([]);
+        const vaultStatic = createMockAssetStaticVaultV2();
+        getStaticMock = jest.fn().mockResolvedValue([vaultStatic]);
+        (vaultInterface as any).getStatic = getStaticMock;
+        vaultDynamic = createMockAssetDynamicVaultV2();
+        const getDynamicMock = jest.fn().mockResolvedValue([vaultDynamic]);
+        (vaultInterface as any).getDynamic = getDynamicMock;
+        vaultsStrategiesMetadataMock.mockResolvedValue([
+          {
+            vaultAddress: "0x001",
+            strategiesMetadata: {
+              name: "strategiesMetadataName",
+              description: "strategiesMetadataDescription",
+              address: "strategiesMetadataAddress",
+              protocols: ["strategiesMetadata"]
+            }
+          }
+        ]);
+        assetsHistoricEarningsMock.mockReturnValue(
```

Comment:
No because in the code we are not awaiting this promise, we expect it to be a promised and not a resolved value. Hence why I had to use `mockReturnValue` and not `mockResolvedValue` in this case.

Reply:
mockResolved value is just an alias for jest.fn().mockImplementation(() => Promise.resolve(value)). So it actually returns a promise 
https://jestjs.io/docs/mock-function-api#mockfnmockresolvedvaluevalue

Reply:
Oh, you're totally right... 🤦 

Refactored with https://github.com/yearn/yearn-sdk/pull/221/commits/34aaf5f57b8fbb282235556a04cfd2d7bd34a694